airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/AirbyteExceptionHandler.java

```diff
@@ -0,0 +1,27 @@
+package io.airbyte.integrations.base;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AirbyteExceptionHandler implements Thread.UncaughtExceptionHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AirbyteExceptionHandler.class);
```

Comment:
Since this is now globally/thread registered, is there ever a case where this might catch an exception before the logger is ready?

Reply:
The logger should be ready by the time this happens, since we're setting this as the handler during IntegrationRunner's init and we [often log before this in the connector](https://github.com/airbytehq/airbyte/blob/f9188590cc61b37e159ec405dc1209f80939eac3/airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchDestination.java#L27-L31)